doc/doc-site/docs/off-chain/dss/data-service-stack.md

```diff
@@ -109,6 +109,47 @@ Finally, to fully shut it down:
 docker compose --file src/docker/compose.dss-core.yaml down
 ```
 
+### Conducting a hot upgrade
+
+If you'd like to upgrade your DSS to the latest version without having to backfill the entire database, you can conduct a "hot upgrade".
+This process leverage's Docker compose' underlying capability to preserve database volumes across environment builds, so that you can recompile binaries without having to sync to chain tip for each new release.
+
+:::note
+The Econia Labs team strives to enable a hot upgrade for every DSS release (relative to the most recent release).
+Occasionally, however, upstream changes (like breaking changes to the Aptos indexing framework) may require backfilling from Move package genesis.
+See the [changelog](./changelog.md) to verify that the latest release supports a hot upgrade.
+:::
+
+1. Shut down the DSS
+
+   ```bash
+   docker compose --file src/docker/compose.dss-core.yaml down
+   ```
+
+1. Clean your Docker containers and image cache:
+
+   ```bash
+   docker ps -aq | xargs docker stop | xargs docker rm
+   docker system prune -af
+   ```
+
+   :::tip
+   This is similar to starting a DSS from scratch, but it preserves the `db` volume.
+   :::
+
+1. Check out the latest stable DSS source code:
+
+   ```bash
+   git checkout dss-stable
+   git submodule update --init --recursive
+   ```
+
+1. Restart the DSS:
+
+   ```bash
+   docker compose --file src/docker/compose.dss-core.yaml up
```

Comment:
How about adding a `docker compose --file src/docker/compose.dss-core.yaml build --no-cache` to make sure everything is correctly rebuilt. In my experience, Docker often doesn't realize something has changed and does not correctly rebuild images.

Reply:
How about `--force-recreate`, added in 94caed2?

https://docs.docker.com/reference/cli/docker/compose/up

Reply:
`create` and `build` are two different concepts. Recreate will recreate a new container from the same image, even if there already is a container created from that image and the image hasn't changed. My command will recreate the image to make sure it corresponds to the latest version.